### DISCUSSION TITLE
How to except self row in unique validation

### DISCUSSION THREAD

**Discussion by mukhsin** [adonisjs/core#2163] [Help] [answered]
Hi all,

I have a table with a unique email column
And I have a validator to check if the input contains a unique email
```
email: schema.string({}, [
  rules.email(),
  rules.unique({
    table: 'users',
    column: 'email',
  }),
])
```

I want the validator to check only unique email except itself so when I updating with the same email as before, it won't throw a validation error for unique email
How can I do that?
I found some options like `where` and `whereNot` besides `table` and `column` but I don't know how to use those

Thanks

**Comment by mukhsin** [comment, marked as answer]
I found the answer in additional constrains in rule.exists
I can use `whereNot` like this
```
email: schema.string({}, [
  rules.email(),
  rules.unique({
    table: 'users',
    column: 'email',
    whereNot: { id: request.param('id) }
  }),
])
```

Thanks

**Comment by victorLessa** [comment]
It doesn't work for me
in the update it continues showing that the data already exists, and even changing all the data this error keeps showing up
